File: simplerpc/src/main/java/com/springmarker/simplerpc/enums/ProtocolType.java
package com.springmarker.simplerpc.enums;

import com.springmarker.simplerpc.core.client.SenderInterface;
import com.springmarker.simplerpc.protocol.net.netty.NettySender;

/**
 * @author Springmarker
 * @date 2018/10/21 22:44
 */

public enum ProtocolType {
    /**
     * Netty协议
     */
    NETTY(NettySender.class);

    ProtocolType(Class<? extends SenderInterface> senderImplClassPath) {
    }
}

File: simplerpc/src/main/kotlin/com/springmarker/simplerpc/enum/ProtocolType.kt
package com.springmarker.simplerpc.enum

import com.springmarker.simplerpc.client.SenderInterface
import com.springmarker.simplerpc.protocol.http.HttpSender

/**
 * @author Frank
 * @date 2018/10/21 22:44
 */
enum class ProtocolType(var senderImplClassPath: Class<out SenderInterface>) {
    HTTP(HttpSender::class.java)
}